src/lib_ccx/hardsubx_decoder.c

```diff
@@ -425,27 +428,42 @@ int hardsubx_process_frames_linear(struct lib_hardsubx_ctx *ctx, struct encoder_
 				progress = (cur_sec*100)/total_sec;
 				activity_progress(progress,cur_sec/60,cur_sec%60);
 
-				if(subtitle_text==NULL)
-					continue;
-				if(!strlen(subtitle_text))
-					continue;
-				char *double_enter = strstr(subtitle_text,"\n\n");
-				if(double_enter!=NULL)
-					*(double_enter)='\0';
-				//subtitle_text = prune_string(subtitle_text);
+				if((!subtitle_text && !prev_subtitle_text) || (subtitle_text && !strlen(subtitle_text) && !prev_subtitle_text)) {
+					prev_end_time = convert_pts_to_ms(ctx->packet.pts, ctx->format_ctx->streams[ctx->video_stream_id]->time_base);
+				}
 
-				end_time = convert_pts_to_ms(ctx->packet.pts, ctx->format_ctx->streams[ctx->video_stream_id]->time_base);
-				if(prev_subtitle_text)
-				{
-					//TODO: Encode text with highest confidence
-					dist = edit_distance(subtitle_text, prev_subtitle_text, strlen(subtitle_text), strlen(prev_subtitle_text));
+				if(subtitle_text) {
+					char *double_enter = strstr(subtitle_text,"\n\n");
+					if(double_enter!=NULL)
+						*(double_enter)='\0';
```

Comment:
Are you sure this is correct? It would terminate the string on the first one, not the second one, so it removes both \n.

Reply:
I haven't added this line. It was there in the source code before I created this pr.